application/src/main/java/org/togetherjava/tjbot/commands/tags/TagManageCommand.java

```diff
@@ -118,9 +124,8 @@ private static OptionalLong parseMessageIdAndHandle(@NotNull String messageId,
     public void onSlashCommand(@NotNull SlashCommandEvent event) {
         Member member = Objects.requireNonNull(event.getMember());
 
-        if (!member.hasPermission(Permission.MESSAGE_MANAGE)) {
-            event.reply(
-                    "Tags can only be managed by users who have the 'MESSAGE_MANAGE' permission.")
+        if (member.getRoles().stream().map(Role::getName).noneMatch(hasRequiredRole)) {
```

Comment:
Maybe extract this logic into a method, possibly `canUserUseTagSystem`.

Reply:
good stuff, done.